### PR TITLE
Script runner perf optimization

### PR DESCRIPTION
**Pull request by tmilar**

### CODE REVIEW

File: packages/buidler-core/src/internal/util/scripts-runner.ts [hunk 1]
```diff
@@ -46,26 +54,54 @@ export async function runScriptWithBuidler(
...
     {
       ...getEnvVariablesMap(buidlerArguments),
       ...extraEnvVars
     }
   );
 }
 
-function getTsNodeArgsIfNeeded() {
+/**
+ * Ensure buidler/register source file path is resolved to compiled JS file
```
[the remark this answers is not in the document]
Comment (tmilar) [reply packages/buidler-core/src/internal/util/scripts-runner.ts]: When running tests on the same module, yes, it resolves to the TS file by default.
However if it's from a different module (ie. placed in node_modules via npm link, or as a dependency), the compiled file will be available.

We can just skip the need of running ts-node at all if we directly use the complied JS file, which will always be available anyways.

File: packages/buidler-core/test/internal/util/scripts-runner.ts [hunk 2]
```diff
@@ -30,54 +35,90 @@ describe("Scripts runner", function() {
...
+    const scriptArgs: string[] = [];
+
+    const runScriptCases = [
+      {
+        scriptPath: "./async-script.js",
+        extraNodeArgs,
+        expectedStatusCode: 0
+      },
+      {
+        scriptPath: "./failing-script.js",
```
[the remark this answers is not in the document]
Comment (tmilar) [reply packages/buidler-core/test/internal/util/scripts-runner.ts]: Yes, I intentionally omitted it just to enforce the same logic as before this refactor. Should I add it as well?

File: packages/buidler-core/src/internal/util/scripts-runner.ts [hunk 3]
```diff
@@ -16,9 +16,17 @@ export async function runScript(
   const { fork } = await import("child_process");
 
   return new Promise((resolve, reject) => {
+    const processExecArgv = process.execArgv.map(arg => {
+      if (arg.toLowerCase().includes("--inspect-brk=")) {
+        // directly use '--inspect' flag, for debuggability
```
[the remark this answers is not in the document]
Comment (tmilar) [reply packages/buidler-core/src/internal/util/scripts-runner.ts]: Added 
https://github.com/nomiclabs/buidler/pull/473/commits/831d2f2e6e4c3288f9dc0a05be28c20f39dc74e6#diff-9c8bc2cdf525a83cc720fc6f48d07719R79

File: packages/buidler-core/src/internal/util/scripts-runner.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (alcuadrado) [reply packages/buidler-core/src/internal/util/scripts-runner.ts]: sounds good! thanks

File: packages/buidler-core/test/internal/util/scripts-runner.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (alcuadrado) [reply packages/buidler-core/test/internal/util/scripts-runner.ts]: It was just a comment. Don't worry.

**Pull request merged**